src/hiccup/compiler.clj

```diff
@@ -237,8 +241,11 @@
   (doall (for [expr content]
            (cond
             (vector? expr) (compile-element expr)
+            (string? expr) `(escape-html ~expr)
```

Comment:
If you know the expression is a literal string, you can just use `(escape-html expr)` and precompute it. Same for the next line.